src/Analyzer/Passes/IfConstantConditionPass.cpp

```diff
@@ -40,18 +44,22 @@ class IfConstantConditionVisitor : public InDepthQueryTreeVisitor<IfConstantCond
         else
             return;
 
+        QueryTreeNodePtr argument_node;
         if (condition_boolean_value)
-            node = function_node->getArguments().getNodes()[1];
+            argument_node = function_node->getArguments().getNodes()[1];
         else
-            node = function_node->getArguments().getNodes()[2];
+            argument_node = function_node->getArguments().getNodes()[2];
+
+        if (node->getResultType()->equals(*argument_node->getResultType()))
```

Comment:
So, otherwise optimization just is not applied to current node, right?

Reply:
Yes.